state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -2302,14 +2345,18 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		};
 
 		if let Some((prewitnessed_deposit_id, used_pools)) = maybe_boost_to_process {
+			let mut total_amount_credited: TargetChainAmount<T, I> = 0u32.into();
 			// Note that ingress fee is not payed here, as it has already been payed at the time
 			// of boosting
 			for boost_tier in used_pools {
 				BoostPools::<T, I>::mutate(asset, boost_tier, |maybe_pool| {
 					if let Some(pool) = maybe_pool {
-						for (booster_id, finalised_withdrawn_amount) in
-							pool.process_deposit_as_finalised(prewitnessed_deposit_id)
-						{
+						let DepositFinalisationOutcomeForPool { unlocked_funds, amount_credited } =
+							pool.process_deposit_as_finalised(prewitnessed_deposit_id);
+
+						total_amount_credited += amount_credited;
```

Comment:
nit: saturating? just to be defensive (though, it would be quite bad if we even got close to saturating here...)